PlagiarismPlugin.inc.php

```diff
@@ -3,29 +3,92 @@
- * Copyright (c) 2003-2021 John Willinsky
+ * Copyright (c) 2003-2024 Simon Fraser University
+ * Copyright (c) 2003-2024 John Willinsky
  * Distributed under the GNU GPL v3. For full terms see the file LICENSE.
  *
  * @brief Plagiarism plugin
  */
 
 import('lib.pkp.classes.plugins.GenericPlugin');
+import('lib.pkp.classes.db.DAORegistry');
 
 class PlagiarismPlugin extends GenericPlugin {
+
+	/**
+	 * Specify a default integration name for iThenticate service
+	 */
+	public const PLUGIN_INTEGRATION_NAME = 'Plagiarism plugin for OJS/OMP/OPS';
+
+	/**
+	 * The default permission of submission primary author's to pass to the iThenticate service
+	 */
+	public const SUBMISSION_AUTOR_ITHENTICATE_DEFAULT_PERMISSION = 'USER';
+
+	/**
+	 * Number of seconds EULA details for a context should be cached before refreshing it
+	 */
+	public const EULA_CACHE_LIFETIME = 60 * 60 * 24;
+
+	/**
+	 * List of valid url components
+	 * 
+	 * @var array
+	 */
+	protected $validRouteComponentHandlers = [
+		'plugins.generic.plagiarism.controllers.PlagiarismWebhookHandler',
+		'plugins.generic.plagiarism.controllers.PlagiarismEulaAcceptanceHandler',
+		'plugins.generic.plagiarism.controllers.PlagiarismIthenticateActionHandler',
+	];
+
 	/**
 	 * @copydoc Plugin::register()
 	 */
 	public function register($category, $path, $mainContextId = null) {
 		$success = parent::register($category, $path, $mainContextId);
+
 		$this->addLocaleData();
+		
+		if (!($success || $this->getEnabled())) {
+			return false;	
+		}
+
+		$context = $mainContextId
+			? Services::get("context")->get($mainContextId)
+			: Application::get()->getRequest()->getContext();
 
-		if ($success && $this->getEnabled()) {
-			HookRegistry::register('submissionsubmitstep4form::execute', array($this, 'callback'));
+		// plugin can not function if the iThenticate service access not available at running context, 
+		if (!$this->isServiceAccessAvailable($context)) {
+			error_log("ithenticate service access not set for context id {$context->getId()}");
+			return false;
 		}
+		
+		HookRegistry::register('submissionsubmitstep4form::display', [$this, 'confirmEulaAcceptance']);
+		HookRegistry::register('submissionsubmitstep4form::execute', [$this, 'submitForPlagiarismCheck']);
+
+		HookRegistry::register('Schema::get::' . SCHEMA_SUBMISSION, [$this, 'addPlagiarismCheckDataToSubmissionSchema']);
+		HookRegistry::register('Schema::get::' . SCHEMA_SUBMISSION_FILE, [$this, 'addPlagiarismCheckDataToSubmissionFileSchema']);
+		HookRegistry::register('Schema::get::' . SCHEMA_CONTEXT, [$this, 'addIthenticateConfigSettingsToContextSchema']);
+		
+		HookRegistry::register('userdao::getAdditionalFieldNames', [$this, 'handleAdditionalEulaConfirmationFieldNames']);
+
+		HookRegistry::register('LoadComponentHandler', [$this, 'handleRouteComponent']);
+
+		HookRegistry::register('editorsubmissiondetailsfilesgridhandler::initfeatures', [$this, 'addActionsToSubmissionFileGrid']);
```

Comment:
https://github.com/asmecher/pkp-plagiarism/commit/b12a018e471c45494771f7e4c57de111ab13b1b3 is an example of using grid features to rearrange the columns. It moves the plagiarism report to the last column.